packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,313 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import { getScope, getSourceCode } from '@typescript-eslint/utils/eslint-utils';
+import type {
+  RuleFix,
+  Scope,
+  SourceCode,
+} from '@typescript-eslint/utils/ts-eslint';
+import * as tsutils from 'ts-api-utils';
+import type { Type } from 'typescript';
+
+import type {
+  MemberExpressionComputedName,
+  MemberExpressionNonComputedName,
+} from '../../../types/src/generated/ast-spec';
```

Comment:
[nit] Other rules don't import `@typescript-eslint/types` package using relative paths. + Additionally `TSESTree` is already imported, I think these types should be accessed from it.

Also, there are `TSESTree.MemberExpressionComputedName` and `MemberExpressionComputedName` in one file. Looks a bit inconsistent :smile: 

Let's refactor it? What do you think?

Reply:
yep, cleaned that up!

Reply:
Maybe a good lint rule internally, as a followup?

Reply:
Filed https://github.com/typescript-eslint/typescript-eslint/pull/8216#discussion_r1472343770

Reply:
Filed https://github.com/typescript-eslint/typescript-eslint/issues/8345